lib/detect-testing-library-utils.ts

```diff
@@ -150,84 +195,92 @@ export function detectTestingLibraryUtils<
-        return true;
-      }
-
-      return !!importedTestingLibraryNode || !!importedCustomModuleNode;
+    const isTestingLibraryImported: IsTestingLibraryImportedFn = () => {
+      return (
+        isAggressiveModuleReportingEnabled() ||
+        !!importedTestingLibraryNode ||
+        !!importedCustomModuleNode
+      );
     };
 
     /**
      * Determines whether filename is valid or not for current file
      * being analyzed based on "testing-library/filename-pattern" setting.
      */
-    const isValidFilename: DetectionHelpers['isValidFilename'] = () => {
+    const isValidFilename: IsValidFilenameFn = () => {
       const fileName = context.getFilename();
       return !!fileName.match(filenamePattern);
     };
 
     /**
      * Determines whether a given node is `get*` query variant or not.
      */
-    const isGetQueryVariant: DetectionHelpers['isGetQueryVariant'] = (node) => {
+    const isGetQueryVariant: IsGetQueryVariantFn = (node) => {
       return /^get(All)?By.+$/.test(node.name);
     };
 
     /**
      * Determines whether a given node is `query*` query variant or not.
      */
-    const isQueryQueryVariant: DetectionHelpers['isQueryQueryVariant'] = (
-      node
-    ) => {
+    const isQueryQueryVariant: IsQueryQueryVariantFn = (node) => {
       return /^query(All)?By.+$/.test(node.name);
     };
 
     /**
      * Determines whether a given node is `find*` query variant or not.
      */
-    const isFindQueryVariant: DetectionHelpers['isFindQueryVariant'] = (
-      node
-    ) => {
+    const isFindQueryVariant: IsFindQueryVariantFn = (node) => {
       return /^find(All)?By.+$/.test(node.name);
     };
 
     /**
      * Determines whether a given node is sync query or not.
      */
-    const isSyncQuery: DetectionHelpers['isSyncQuery'] = (node) => {
+    const isSyncQuery: IsSyncQueryFn = (node) => {
       return isGetQueryVariant(node) || isQueryQueryVariant(node);
     };
 
     /**
      * Determines whether a given node is async query or not.
      */
-    const isAsyncQuery: DetectionHelpers['isAsyncQuery'] = (node) => {
+    const isAsyncQuery: IsAsyncQueryFn = (node) => {
       return isFindQueryVariant(node);
     };
 
-    const isCustomQuery: DetectionHelpers['isCustomQuery'] = (node) => {
+    const isCustomQuery: IsCustomQueryFn = (node) => {
       return (
         (isSyncQuery(node) || isAsyncQuery(node)) &&
         !ALL_QUERIES_COMBINATIONS.includes(node.name)
       );
     };
 
     /**
-     * Determines whether a given node is async util or not.
+     * Determines whether a given node is a valid async util or not.
+     *
+     * A node will be interpreted as a valid async util based on two conditions:
+     * the name matches with some Testing Library async util, and the node is
+     * coming from Testing Library module.
+     *
+     * The latter depends on Aggressive
+     * module reporting: if enabled, then it doesn't matter from
```

Comment:
The line break seems weird, is it intended?

Reply:
Not really. It probably go split in two lines after some rewording. I'll fix it.